components/scream/src/physics/rrtmgp/atmosphere_radiation.cpp

```diff
@@ -531,6 +534,40 @@ void RRTMGPRadiation::run_impl (const int dt) {
     gas_concs.set_vmr(name, tmp2d);
   }
 
+  // Set layer cloud fraction.
+  //
+  // If not doing subcolumn sampling for mcica, we want to make sure we use grid-mean
+  // condensate for computing cloud optical properties, because we are assuming the
+  // entire column is completely clear or cloudy. Thus, in this case we want to set
+  // cloud fraction to 0 or 1. Note that we could choose an alternative threshold
+  // criteria here, like qc + qi > 1e-5 or something.
+  //
+  // If we *are* doing subcolumn sampling for MCICA, then keep cloud fraction as input
+  // from cloud fraction parameterization, wherever that is computed.
+  auto do_subcol_sampling = m_do_subcol_sampling;
+  if (not do_subcol_sampling) {
+    const auto policy = ekat::ExeSpaceUtils<ExeSpace>::get_default_team_policy(m_ncol, m_nlay);
+    Kokkos::parallel_for(policy, KOKKOS_LAMBDA(const MemberType& team) {
+      const int i = team.league_rank();
+      Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlay), [&] (const int& k) {
+        if (d_cldfrac_tot(i,k) > 0) {
+          cldfrac_tot(i+1,k+1) = 1;
+        } else {
+          cldfrac_tot(i+1,k+1) = 0;
```

Comment:
could we just initialize `cldfrac_tot` to 0 everywhere and then not need the `else` part of this conditional?

Reply:
Yeah we probably could! I can do that but it does mean a separate kernel.

Reply:
It's not more involved, it's just a `memset(cldfrac_tot, 0.0)`, but my comment was that I think in either case this probably launches a kernel under the hood to do the "work" of setting to zero, especially since this is a device array/view.